std/typecons.d

```diff
@@ -149,25 +157,37 @@ public:
     ~this()
     {
         debug(Unique) writeln("Unique destructor of ", (_p is null)? null: _p);
-        if (_p !is null) delete _p;
-        _p = null;
+        release();
     }
-    /** Returns whether the resource exists. */
-    @property bool isEmpty() const
-    {
-        return _p is null;
-    }
-    /** Transfer ownership to a $(D Unique) rvalue. Nullifies the current contents. */
-    Unique release()
+
+    /** Frees the underlying $(D RefT) and nulls it */
+    void release()
     {
         debug(Unique) writeln("Release");
-        auto u = Unique(_p);
-        assert(_p is null);
-        debug(Unique) writeln("return from Release");
-        return u;
+        if (_p !is null)
+        {
+            import core.memory : GC;
+
+            destroy(_p);
+            GC.free(cast(void*)_p);
+            _p = null;
+        }
     }
-    /** Forwards member access to contents. */
-    RefT opDot() { return _p; }
+
+    /**
+    Returns the underlying $(D RefT) for use by non-owning code.
+
+    The holder of a $(D Unique!T) is the <em>owner</em> of that $(D T).
+    For code that does not own the resource (and therefore does not affect
+    its life cycle), pass a plain old reference.
+    */
+    RefT get() { return _p; }
+
+    /**
+    Allows you to dereference the underlying $(D RefT)
+    and treat it like a $(D RefT) in other ways (such as comparing to null)
+    */
+    alias _p this;
```

Comment:
Did you take the time to watch the linked section of the talk? D _can_ (unlike C++) do compile-time checks for safety using `@safe`, but since it can't verify life cycles at compile time (a la Rust), I don't know 

If we want to mark this as unsafe, that's fine, but I still strongly believe it's necessary. What's the alternative? Passing `ref Unique!T` around? That's problematic for the reasons listed above.